extension/source/QuillPage/components/ViewSecretPhrasePanel.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { browser } from 'webextension-polyfill-ts';
+import Button from '../../components/Button';
+import Range from '../../helpers/Range';
+import QuickColumn from './QuickColumn';
+import QuickRow from './QuickRow';
+
+const ViewSecretPhrasePanel: React.FunctionComponent = () => {
+  const [expanded, setExpanded] = React.useState(false);
+
+  const classes = ['view-secret-phrase-panel'];
+
+  if (expanded) {
+    classes.push('expanded');
+  }
+
+  const secretWords = [
+    'Potato',
+    'Velvet',
+    'Keen',
+    'Water',
+    'Travel',
+    'Pill',
+    'Book',
+    'Photo',
+    'Image',
+    'Space',
+    'Pause',
+    'Power',
+  ];
+
+  return (
+    <div className={classes.join(' ')}>
+      <div className="instructions-text">
+        <h3>
+          Congratulations!
+          <br />
+          You have created a wallet.
+        </h3>
+        <p>
+          Below is your secret recovery phrase, which you will need when
+          restoring your wallets should you lose access.
+        </p>
+      </div>
+      <div className="secret-phrase-box">
+        {Range(4).map((i) => (
+          <QuickRow key={`row${i}`}>
+            {Range(3).map((j) => (
+              <QuickColumn key={`column${j}`}>
+                {3 * i + j + 1}. {secretWords[3 * i + j]}
+              </QuickColumn>
+            ))}
+          </QuickRow>
+        ))}
+      </div>
+      <div className="show-box">
+        <div style={{ display: 'inline-block' }}>
+          <Button
+            onPress={() => setExpanded(true)}
+            highlight={false}
+            icon={{
+              src: browser.runtime.getURL('assets/eye.svg'),
+              px: 19,
+            }}
+          >
+            Show secret phrase
+          </Button>
+        </div>
+      </div>
```

Comment:
```suggestion
      <QuickRow>
        <div style={{ display: 'inline-block' }}>
          <Button
            onPress={() => setExpanded(!expanded)}
            highlight={false}
            icon={{
              src: browser.runtime.getURL('assets/eye.svg'),
              px: 19,
            }}
          >
            {expanded ? "Hide secret phrase" : "Show secret phrase"}
          </Button>
        </div>
        {expanded && <Button
            onPress={() => {}}
            highlight={true}
            icon={{
              src: browser.runtime.getURL('assets/arrow-small.svg'),
              px: 19,
            }}
          >
            Review secret phrase
          </Button>}
        </QuickRow>
```

Can use a single button for showing/hiding the box. Might need to change the CSS as well. 
`display:none` in CSS still renders an invisible component

Reply:
> `display:none` in CSS still renders an invisible component

[I don't think that's true.](https://stackoverflow.com/questions/11757016/performance-differences-between-visibilityhidden-and-displaynone)

Does this change your view? Either way I feel these are premature optimizations and it's probably better to move forward than iron them out.

Reply:
> Does this change your view?

No, Not really. Also the discussion is from 2012.
The element must atleast be present in the DOM for CSS to be applied on it. If there is no element then there cant be any style.

<img width="1717" alt="Screenshot 2022-01-14 at 5 29 19 PM" src="https://user-images.githubusercontent.com/23727056/149512563-58296848-ca33-41f4-bd51-607b737dba83.png">

<img width="1712" alt="Screenshot 2022-01-14 at 5 30 29 PM" src="https://user-images.githubusercontent.com/23727056/149512600-54a7854c-9c89-4b0e-a62e-efe8afb8d0e3.png">

> Either way I feel these are premature optimizations
 
Yeah I do agree. However what i feel that once the code base grows big, these things will be hard to find. Better to take action on smaller things than leave them for later. 

eg - (this)[https://github.com/jzaki/bls-wallet/pull/88#discussion_r780971630] was a big enough change which could be delayed for later.

Reply:
Ah I see you meant a different kind of 'render'.

I'm not convinced it matters but I also don't mind either way. Fixed.